### ISSUE TITLE
Resource limit error again

### ISSUE THREAD

**Issue by Seraphim-AT** [airsdk/Adobe-Runtime-Support#2388]
There was similar issue in the past, here it is: https://github.com/airsdk/Adobe-Runtime-Support/issues/245
But now it appeared again.

AIR - 50.0.1.1, 50.0.1.3, 50.1.1.2, so it appears in any 50 namespace (http://ns.adobe.com/air/application/50.x)
Mac OS

Using big textures causes an error:

`Error #3691: Resource limit for this resource type exceeded.`

But if I change app namespace to 33 (like http://ns.adobe.com/air/application/33.1, even if I use latest SDK ver 50.1.1.2) - error disappears.

**Comment by spAds2** [comment]
Can confirm that this suddenly randomly happens since AIR 50.* No issues for 5+ years before that.

**Comment by ajwfrost** [comment]
I've just looked at the code here ... there was actually an "ignore resource limits" flag that we put in, but that's only set up for when the API level was 33.1. From the history, a change was put in on 16-Jan-2020 which ignored the limits for an API version below 33.1, but then we expanded the coverage of the flag to cover other limits (vertex buffer too big etc). This change happened 7th Oct 2022..

At this point, the flag was added to the Context3D API so that it could be manually set up by developers who wanted to use it, but the default value still reflected what had been done previously i.e. an API level below 33.1 meant that it would have the limits, an API level of 33.1 exactly means it would ignore limits (unless told to enforce them).

So basically -> to get AIR 50 to work the same as AIR 33.1, you need to use the new AS3 API for this:
https://airsdk.dev/reference/actionscript/3.0/flash/display3D/Context3D.html#ignoreResourceLimits

Are you able to set that flag to `true` and check whether it helps? AFAIK it should then no longer be possible to get that error `#3691` thrown if the value there is true..

thanks

**Comment by spAds2** [comment]
I'll try it.
For us it mostly appears when we run our integration test that speeds through the app. Somewhere disposing seems to not always kick in correctly.

For Starling users:
`myStarling.stage3D.context3D.ignoreResourceLimits = true;`

**Comment by Seraphim-AT** [comment]
solution works for me in Starling version. thanks a lot!

**Comment by Adolio** [comment]
@Seraphim-AT If you're happy with the provided answer / solution, can you close this issue please? Thanks :)

**Comment by 2jfw** [comment]
Just encountered that issue in `Feathers` : `ImageLoader` @ `replaceBitmapDataTexture`, where that `Error` was (silently) caught, having lead to a rather extensive search for the reason.

Reason (catch-block):
```
try
{
	this._texture = Texture.empty(bitmapData.width / this._scaleFactor,
		bitmapData.height / this._scaleFactor, true, false, false,
		this._scaleFactor, this._textureFormat);
}
catch(error:Error)
{
	this.cleanupTexture();
	this.invalidate(INVALIDATION_FLAG_DATA);
	this.dispatchEventWith(starling.events.Event.IO_ERROR, false, new IOErrorEvent(IOErrorEvent.IO_ERROR, false, false, error.toString()));
	return;
}
```

Solution from https://github.com/airsdk/Adobe-Runtime-Support/issues/2388#issuecomment-1410526512 works:
`myStarling.stage3D.context3D.ignoreResourceLimits = true;`
👍

Many thanks! 🍻

**Comment by PippoApps** [comment]
There is another strange error.
If the app is moved to another screen, before or after creating a Context3D, GPU limits are set to low specs.
An app that works perfectly up to 4GB of video ram, if moved to another screen, crashes at 512MB.

**Comment by PippoApps** [comment]
Please check: https://github.com/airsdk/Adobe-Runtime-Support/issues/3731#issuecomment-2765994712
